src/autopas/containers/verletClusterLists/ClusterTowerBlock2D.h

```diff
@@ -0,0 +1,437 @@
+#include "autopas/containers/verletClusterLists/ClusterTower.h"
+#include "autopas/utils/ArrayMath.h"
+
+namespace autopas::internal {
+
+/**
+ * Class to manage the grid of towers for the Verlet Cluster Lists container.
+ * This also includes the geometric shape of the container.
+ *
+ * @tparam Particle
+ */
+template <class Particle>
+class ClusterTowerBlock2D : public CellBorderAndFlagManager {
+ public:
+  /**
+   * Constructor.
+   *
+   * @note This constructor does not initialize the tower grid! For this, the number of particles is required.
+   *
+   * @param boxMin
+   * @param boxMax
+   * @param interactionLength
+   */
+  ClusterTowerBlock2D(const std::array<double, 3> &boxMin, const std::array<double, 3> &boxMax,
+                      double interactionLength)
+      : _interactionLength(interactionLength),
+        _boxMin(boxMin),
+        _boxMax(boxMax),
+        _haloBoxMin{utils::ArrayMath::subScalar(boxMin, interactionLength)},
+        _haloBoxMax{utils::ArrayMath::addScalar(boxMax, interactionLength)} {}
+  /**
+   * Start iterator over towers.
+   * @return
+   */
+  typename std::vector<ClusterTower<Particle>>::iterator begin() { return _towers.begin(); }
+  /**
+   * Start iterator over towers.
+   * @return
+   */
+  typename std::vector<ClusterTower<Particle>>::const_iterator begin() const { return _towers.begin(); }
+  /**
+   * End iterator over towers.
+   * @return
+   */
+  typename std::vector<ClusterTower<Particle>>::iterator end() { return _towers.end(); }
+  /**
+   * End iterator over towers.
+   * @return
+   */
+  typename std::vector<ClusterTower<Particle>>::const_iterator end() const { return _towers.end(); }
+
+  /**
+   * Access operator for towers.
+   * @param i 1D tower index
+   * @return
+   */
+  ClusterTower<Particle> &operator[](size_t i) { return _towers[i]; }
+
+  /**
+   * Access operator for towers.
+   * @param i 1D tower index
+   * @return
+   */
+  const ClusterTower<Particle> &operator[](size_t i) const { return _towers[i]; }
+
+  /**
+   * Return the number of towers.
+   * @return
+   */
+  size_t size() const { return _towers.size(); }
+
+  /**
+   * Resize the internal grid and storage.
+   * @param towerSideLength
+   * @param towersPerDim
+   */
+  void resize(const std::array<double, 2> &towerSideLength, const std::array<size_t, 2> &towersPerDim) {
+    using namespace autopas::utils::ArrayMath::literals;
+    using autopas::utils::ArrayMath::ceil;
+    using autopas::utils::ArrayUtils::static_cast_copy_array;
+    _towerSideLength = towerSideLength;
+    _towerSideLengthReciprocal = 1. / towerSideLength;
+    const auto numTowersPerInteractionLength2D =
+        static_cast_copy_array<int>(ceil(_interactionLength / _towerSideLength));
+    // sanity check
+    if (numTowersPerInteractionLength2D[0] != numTowersPerInteractionLength2D[1]) {
+      AutoPasLog(WARN, "Number of towers per interaction length differs in X vs Y direction! {}",
```

Comment:
Honestly, I have no idea. The thing is, that the container was built with the assumption that they don't differ, and I did not look into the details where this might become a problem. I just thought this is something we might want to warn about to avoid silent bugs.-> 10e694129